### DISCUSSION TITLE
is there a MACRO definition tell target board core frequency?

### DISCUSSION THREAD

**Discussion by xiongyu0523** [stm32duino/Arduino_Core_STM32#1457] [Q&A] [answered]
Previously I use F_CPU for other architecture (samd), but on stm32 core, the F_CPU is SystemCoreClock, which is not predefined value.  my question is that is there a MACRO definition tell target board core frequency?

I need the value for compiler to pre-calculate the proper clock cycle for systick.

**Comment by ABOSTM** [comment, marked as answer]
Hi @xiongyu0523,
Thanks to PR: https://github.com/stm32duino/Arduino_Core_STM32/pull/613
you can now override definition of F_CPU with a predefine value of your choice (of course it must match the clock value which will be configured).
